File: app/src/main/cpp/wn-steam-client/src/pb/cmsg_client_change_status.cpp
#include "wn_steam/pb/cmsg_client_change_status.h"

#include "wn_steam/proto_wire.h"

namespace wn_steam::pb {

std::vector<uint8_t> CMsgClientChangeStatus::serialize() const {
    std::vector<uint8_t> out;
    proto::Writer w(out);
    // Force-emit: persona_state 0 (Offline) is a meaningful value.
    w.uint32_field_force(1, persona_state);
    return out;
}

}  // namespace wn_steam::pb

File: app/src/main/cpp/wn-steam-client/rust/src/pb/cmsg_client_change_status.rs
use crate::proto_wire::Writer;

#[derive(Clone, Debug, Default, Eq, PartialEq)]
pub struct CMsgClientChangeStatus {
    pub persona_state: u32,
    pub player_name: String,
    pub persona_set_by_user: bool,
    pub need_persona_response: bool,
}

impl CMsgClientChangeStatus {
    pub fn serialize(&self) -> Vec<u8> {
        let mut out = Vec::new();
        let mut w = Writer::new(&mut out);
        w.uint32_field_force(1, self.persona_state);
        w.string_field(2, &self.player_name);
        if self.persona_set_by_user {
            w.bool_field(5, true);
        }
        if self.need_persona_response {
            w.bool_field(7, true);
        }
        out
    }
}
